aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -211,7 +211,7 @@ Tensor batch_norm_elementwise_backward_train(
     auto weight_nd = weight.defined() ? as_nd(weight) :
         at::scalar_tensor(1.0, input.options().dtype(mean.scalar_type()));
 
-    Tensor grad_input = at::empty(input.sizes(), grad_out.options());
+    Tensor grad_input = at::empty(input.sizes(), grad_out.options()).to(input.suggest_memory_format());
```

Comment:
add this to the options() argument of empty. No need for a full copy!

Reply:
my bad, fixed!